### PR TITLE
Add unit tests for azure race conditions.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake
/kind cleanup

**What this PR does / why we need it**:
Add unit tests for the concurrent update, etag mismatch and something else.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/area provider/azure
/priority important-soon

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go [hunk 1]
```diff
@@ -2186,3 +2189,130 @@ func TestEnsureBackendPoolDeleted(t *testing.T) {
...
+	backendpoolAddressIDs := []string{testLBBackendpoolID0, testLBBackendpoolID1, testLBBackendpoolID2}
+	testFunc := make([]func() error, 0)
+	for _, id := range backendpoolAddressIDs {
+		id := id
+		testFunc = append(testFunc, func() error {
+			return ss.EnsureBackendPoolDeleted(&v1.Service{}, id, testVMSSName, backendAddressPools)
+		})
+	}
+	errs := utilerrors.AggregateGoroutines(testFunc...)
+	assert.Equal(t, 1, len(errs.Errors()))
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go:2317]: could you add one more assert for the error?

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go [hunk 2]
```diff
@@ -2186,3 +2189,130 @@ func TestEnsureBackendPoolDeleted(t *testing.T) {
...
+				BackendIPConfigurations: &[]network.InterfaceIPConfiguration{
+					{
+						Name: to.StringPtr("ip-1"),
+						ID:   to.StringPtr("/subscriptions/sub/resourceGroups/rg1/providers/Microsoft.Compute/virtualMachineScaleSets/vmss/virtualMachines/0/networkInterfaces/nic"),
+					},
+				},
+			},
+		},
+	}
+	expectedVMSS := compute.VirtualMachineScaleSet{
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_test.go:2236]: any possible to add some helper funtions for generating VirtualMachineScaleSet and VirtualMachineScaleSetVM? They could be reused in other test cases

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff_test.go [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+
+	tests := []struct {
+		vmss        compute.VirtualMachineScaleSet
+		clientErr   *retry.Error
+		expectedErr *retry.Error
+	}{
+		{
+			clientErr:   &retry.Error{HTTPStatusCode: http.StatusInternalServerError},
+			expectedErr: &retry.Error{HTTPStatusCode: http.StatusInternalServerError},
+		},
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff_test.go:476]: add two more cases for throttling and rate limited?

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff_test.go [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+	az := GetTestCloud(ctrl)
+	mockLBClient := az.LoadBalancerClient.(*mockloadbalancerclient.MockInterface)
+	mockLBClient.EXPECT().List(gomock.Any(), az.ResourceGroup).Return(nil, &retry.Error{HTTPStatusCode: http.StatusInternalServerError})
+
+	pips, err := az.ListLB(&v1.Service{})
+	assert.Equal(t, fmt.Errorf("Retriable: false, RetryAfter: 0s, HTTPStatusCode: 500, RawError: <nil>"), err)
+	assert.Empty(t, pips)
+}
+
+func TestListPIP(t *testing.T) {
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff_test.go:291]: could we add both positive and negative cases for PIP operations? (same for the below)

Comment (nilo19) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_backoff_test.go:291]: The positive cases have been covered somewhere else.

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by saschagrunert** [comment]
Hey @nilo19, may I ask here too to change the PR release notes block to just `None`?

**Pull request merged**